src/test/java/com/buddy/tutorial/controller/CurrencyControllerTest.java

```diff
@@ -0,0 +1,69 @@
+package com.buddy.tutorial.controller;
+
+import com.buddy.tutorial.model.CurrencyInfo;
+import com.buddy.tutorial.model.ResponseModel;
+import com.buddy.tutorial.service.CurrencyService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class CurrencyControllerTest {
+
+    @Mock
+    private CurrencyService currencyService;
+
+    @InjectMocks
+    private CurrencyController controller;
+
+    @BeforeEach
+    void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void getAllCurrency_Success() {
+
+        CurrencyInfo currencyInfo = new CurrencyInfo();
```

Comment:
same CurrencyInfo  is used across multiple methods. move to a common method.
same applies in service test class as well